From c96a090cb80ab98c687367521d4bda8441b8bf23 Mon Sep 17 00:00:00 2001
From: Ricardo Cerqueira <cyanogenmod@cerqueira.org>
Date: Sun, 19 Jan 2014 19:07:46 +0000
Subject: [PATCH 19/51] Allow library to be used by apps directly

Change-Id: Ib74be626471d73817d644c7777a2b18a32712f8a

Author: Adrian DC <radian.dc@gmail.com>
Date:   Thu Mar 8 22:46:55 2018 +0100
fm: Resolve required dependency qcom.fmradio.xml linkage

 * LOCAL_REQUIRED_MODULES should be used instead of
    LOCAL_ADDITIONAL_DEPENDENCIES to link a module

 * fm/Android.mk:
    error: qcom.fmradio: LOCAL_ADDITIONAL_DEPENDENCIES
    must only contain paths (not module names)

Change-Id: Ic0bebf752101f5d16abc52e085bae4acd0b0c61c
---
 Android.mk                    | 12 ++++++++++++
 qcom/fmradio/qcom.fmradio.xml |  5 +++++
 2 files changed, 17 insertions(+)
 create mode 100644 qcom/fmradio/qcom.fmradio.xml

diff --git a/Android.mk b/Android.mk
index 2c77dc0c..700ec4f4 100644
--- a/Android.mk
+++ b/Android.mk
@@ -12,8 +12,20 @@ LOCAL_JNI_SHARED_LIBRARIES := libqcomfm_jni
 LOCAL_SHARED_LIBRARIES := liblog
 LOCAL_MODULE:= qcom.fmradio
 
+LOCAL_REQUIRED_MODULES := qcom.fmradio.xml
+
 include $(BUILD_JAVA_LIBRARY)
 
+include $(CLEAR_VARS)
+
+LOCAL_MODULE := qcom.fmradio.xml
+LOCAL_SRC_FILES := qcom/fmradio/$(LOCAL_MODULE)
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := ETC
+LOCAL_MODULE_PATH := $(TARGET_OUT)/etc/permissions
+
+include $(BUILD_PREBUILT)
+
 ifeq ($(BOARD_HAS_QCA_FM_SOC), "cherokee")
 LOCAL_CFLAGS += -DFM_SOC_TYPE_CHEROKEE
 endif
diff --git a/qcom/fmradio/qcom.fmradio.xml b/qcom/fmradio/qcom.fmradio.xml
new file mode 100644
index 00000000..6a55cc66
--- /dev/null
+++ b/qcom/fmradio/qcom.fmradio.xml
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<permissions>
+    <library name="qcom.fmradio"
+            file="/system/framework/qcom.fmradio.jar" />
+</permissions>
-- 
2.17.1

